library/Ivoz/Provider/Domain/Model/CallCsvScheduler/CallCsvSchedulerDto.php

```diff
@@ -90,12 +93,121 @@ public function getCompanyType()
     }
 
     /**
-     * Remove this as soon as klear is dead
+     * TODO: Remove this as soon as klear is dead
      */
     public function setCompanyType(string $type = null)
     {
+        $this->type = $type;
         if (is_null($type)) {
             $this->setCompany(null);
+            $this->setUser(null);
+            $this->setFax(null);
+            $this->setFriend(null);
+            $this->setRetailAccount(null);
+            $this->setResidentialDevice(null);
+        } elseif ($type === 'vpbx') {
+            $this->setRetailAccount(null);
+            $this->setResidentialDevice(null);
+        } elseif ($type === 'retail') {
+            $this->setUser(null);
+            $this->setFax(null);
+            $this->setFriend(null);
+            $this->setResidentialDevice(null);
+        } elseif ($type === 'residential') {
+            $this->setUser(null);
+            $this->setFax(null);
+            $this->setFriend(null);
+            $this->setRetailAccount(null);
+        }
+
+        return $this;
+    }
+
+    /**
+     * TODO: Remove this as soon as klear is dead
+     */
+    public function getVpbxId()
+    {
+        return $this->getCompanyId();
+    }
+
+    /**
+     * TODO: Remove this as soon as klear is dead
+     */
+    public function setVpbxId($id)
+    {
+        if ($this->type === 'vpbx') {
+            $this->setCompanyId($id);
+        }
+
+        return $this;
+    }
+
+    /**
+     * TODO: Remove this as soon as klear is dead
+     */
+    public function getRetailId()
+    {
+        return $this->getCompanyId();
+    }
+
+    /**
+     * TODO: Remove this as soon as klear is dead
+     */
+    public function setRetailId($id)
+    {
+        if ($this->type === 'retail') {
+            $this->setCompanyId($id);
+        }
+
+        return $this;
+    }
+
+    /**
+     * TODO: Remove this as soon as klear is dead
+     */
+    public function getResidentialId()
+    {
+        return $this->getCompanyId();
+    }
+
+    /**
+     * TODO: Remove this as soon as klear is dead
+     */
+    public function setResidentialId($id)
+    {
+        if ($this->type === 'residential') {
+            $this->setCompanyId($id);
         }
+
+        return $this;
+    }
+
+    /**
+     * TODO: Remove this as soon as klear is dead
+     */
+    public function getEndpointType()
```

Comment:
Are these strings only for lists info or are actually used for filtering BillableCalls? If the Second, then this should also be BillableCallsInterface constants

Reply:
This is just a made up field witch exist just to allow some visual filters on new/edit screens.